pkg/distributor/distributor_test.go

```diff
@@ -215,15 +215,39 @@ func TestDistributorPushQuery(t *testing.T) {
 	}
 }
 
-func prepare(t *testing.T, numIngesters, happyIngesters int, shardByAllLabels bool) *Distributor {
+func TestSlowQueries(t *testing.T) {
+	nameMatcher := mustEqualMatcher("__name__", "foo")
+	nIngesters := 3
+	for _, shardByAllLabels := range []bool{true, false} {
+		for happy := 0; happy <= nIngesters; happy++ {
+			var expectedErr error
+			if nIngesters-happy > 1 {
+				expectedErr = errFail
```

Comment:
I'm still building my unit test muscles, but does this imply that this condition may not be true at times, and thus `expectedErr` should be `nil` when we `assert.Equal` it later?

I would expect a test to check for the same behavior each time, and thus just "expect" a non-`nil` error every time yea?

Reply:
Since we are looping across the number of happy (working) ingesters, some requests are going to fail, and some are going to succeed. In this case we have 3 ingesters, and a replication factor of 3, so if more than 1 ingester is unhappy then our request is going to fail. If 0 or 1 ingester is unhappy, then we do not expect any error.

Reply:
Ooooh right right. So it's deterministic in the end, just not per ingester. :+1: Thanks!